runtime/src/snapshot_utils.rs

```diff
@@ -1769,6 +1784,28 @@ fn rebuild_bank_from_snapshots(
     Ok(bank)
 }
 
+/// Verify that the snapshot's slot deltas are not corrupt/invalid
+fn verify_slot_deltas(
+    slot_deltas: &[BankSlotDelta],
+    bank_slot: Slot,
+) -> std::result::Result<(), VerifySlotDeltasError> {
+    for (slot, is_root, _) in slot_deltas {
+        // all entries should be roots
+        if !is_root {
+            return Err(VerifySlotDeltasError::SlotIsNotRoot(*slot));
+        }
+
+        // all entries should be for slots less than or equal to the bank's slot
+        if *slot > bank_slot {
```

Comment:
:+1:

Reply:
could you also verify lower bound of `*slot` with picking a valid offset after reading the relevant bank code?

Reply:
>  Is there a minimum slot check somewhere else that you were thinking of and could point me to?

@brooksprumo (sorry for delayed review...)

hmm, there should be none atm. that's the reason snapshot code isn't well guarded in general. and you're helping remedying the situation, sounds cool. :)

come to think of it, you can compare against the slot history sysvar. For sample, see `adjust_lockouts_after_replay` (btw, one of my proud robust code is here, lol. fyi, it has unfixed really special case insignificant bug still...):

https://github.com/solana-labs/solana/blob/ad0acaa6fa5b6c113f916b5b72f74f199bbef612/core/src/consensus.rs#L1162

Referencing sysvars for snapshot sanitization is the general direction, which I'd like to see. And, putting more deta onto chain as new sysvars as well. so that we have moar trusted (voted) on chain data for internal bank state sanitization. maybe eventually all. :)

Using the slot history sysvar, you can compute precise block height, thus lower bound in terms of slot number by looping over the sysvar  `MAX_CACHE_ENTRIES` times.

Also, you can now validate whether the given slot delta for a slot is actually for a skipped block or not. And more, you can detect any missing slot delta which should exist actually according to the slot history sysvar.

> did not see a minimum slot check. This makes sense to me, as slots are not guaranteed to be rooted, right?

hmm, there is no minimum slot check for lack of manpower. and it should be unrelated to another fact that slots aren't rooted always. What links these things together in your mind?